File: server/DocumentsKM/Models/AttachedDoc.cs
using System.ComponentModel.DataAnnotations;

namespace DocumentsKM.Models
{
    // Прилагаемый документ марки
    public class AttachedDoc
    {
        [Key]
        public int Id { get; set; }

        // Марка
        [Required]
        public virtual Mark Mark { get; set; }

        // Обозначение
        [Required]
        [MaxLength(40)]
        public string Designation { get; set; }

        // Наименование
        [Required]
        [MaxLength(255)]
        public string Name { get; set; }

        // Примечание
        [MaxLength(255)]
        public string Note { get; set; }
    }
}

File: client/src/model/AttachedDoc.ts
interface IAttachedDoc {
    id: number
    designation: string
    name: string
    note: string
}

class AttachedDoc {
	id: number
    designation: string
    name: string
    note: string

	constructor(obj?: IAttachedDoc) {
        this.id = (obj && obj.id) || 0
        this.designation = (obj && obj.designation) || ''
		this.name = (obj && obj.name) || ''
		this.note = (obj && obj.note) || ''
	}
}

export default AttachedDoc
